core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -907,8 +911,9 @@ private[kafka] class Processor(val id: Int,
             else {
               val nowNanos = time.nanoseconds()
               if (channel.serverAuthenticationSessionExpired(nowNanos)) {
-                channel.disconnect()
-                debug(s"Disconnected expired channel: $channel : $header")
+                // be sure to decrease connection count and drop any in-flight responses
+                close(channel.id)
+                error(s"Disconnected expired channel: $channel : $header")
```

Comment:
Is this really an error?

Reply:
@hachikuji thoughts?  This change came from your original commit.

Reply:
I might have changed this while debugging at some point. I agree it probably should not be an error. Do you think it merits info level?

Reply:
`close(connectionId)` emits `debug(s"Closing selector connection $connectionId")`, so we get that message now that we are invoking that method instead of `channel.disconnect()`.  If we want to log anything here then at a minimum we should provide additional context at the same `debug` level (i.e. not only is the connection being closed, but it is being closed for the specific reason that the connection expired).  The question is: if we want to provide additional information here, should it have the more-visible `info` level?  I do not yet have a good sense of when different log levels are used, so at this point all I can justifiably say is that the original reviewer of this code concluded that a `debug` message was acceptable here (I went back to the [original PR](https://github.com/apache/kafka/pull/5582) and looked at the comments, and there were non related to that specific line).

In the interest of making progress I will revert this code to the original `debug` level in 24 hours if I don't hear anything to the contrary from someone else -- in which case we are providing additional context at the same `debug` level.

Reply:
Sounds fine to me.